app/src/main/java/org/p2p/wallet/history/ui/history/HistoryFragment.kt

```diff
@@ -94,19 +91,21 @@ class HistoryFragment :
         with(binding.layoutHistoryList) {
             shimmerView.root.isVisible = state == PagingState.InitialLoading
             refreshLayout.isVisible = state != PagingState.InitialLoading
-            errorStateLayout.root.isVisible = state is PagingState.Error
+            errorStateLayout.root.isVisible = state is PagingState.Error && adapter.isEmpty()
             emptyStateLayout.root.isVisible = state == PagingState.Idle && adapter.isEmpty()
-            historyRecyclerView.isVisible =
-                (state == PagingState.Idle && !adapter.isEmpty()) || state == PagingState.Loading
+            historyRecyclerView.isVisible = isHistoryItemsVisible(state)
```

Comment:
Can we clean this code by making `when`?
```
when (state) {
  is PagingState.Idle -> TODO()
  is PagingState.Loading-> TODO()
  is PagingState.Error -> TODO()
  etc.
}
```

Reply:
ok will try to rewrite